core/src/main/java/org/apache/metamodel/query/parser/SelectItemParser.java

```diff
@@ -113,10 +110,10 @@ public SelectItem findSelectItem(String expression) throws MultipleSelectItemsPa
         final int startParenthesis = expression.indexOf('(');
         if (startParenthesis > 0 && expression.endsWith(")")) {
             String functionName = expression.substring(0, startParenthesis);
-            function = FunctionType.get(functionName.toUpperCase());
+            function = FunctionTypeFactory.get(functionName.toUpperCase());
             if (function != null) {
                 expression = expression.substring(startParenthesis + 1, expression.length() - 1).trim();
-                if (function == FunctionType.COUNT && "*".equals(expression)) {
+                if (function instanceof CountAggregateFunction && "*".equals(expression)) {
```

Comment:
We actually have SelectItem.isCountAllItem(...) to do this - maybe better to reuse.